lib/grizzly/zipgateway/monitor.ex

```diff
@@ -0,0 +1,30 @@
+defmodule Grizzly.ZIPGateway.Monitor do
+  @moduledoc false
+
+  use GenServer
+
+  require Logger
+
+  @spec start_link([]) :: GenServer.on_start()
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [])
+  end
+
+  @impl GenServer
+  def init(_) do
+    Process.flag(:trap_exit, true)
+    :telemetry.execute([:grizzly, :zip_gateway, :start], %{}, %{})
+    {:ok, %{}}
+  end
+
+  @impl GenServer
+  def handle_info({:EXIT, _from, _reason}, state) do
+    {:noreply, state}
+  end
+
+  @impl GenServer
+  def terminate(_, _) do
+    Logger.warning("[Grizzly] Z/IP Gateway process exited")
```

Comment:
I'm wondering if this should be an alert level. If this happens and it wasn't triggered by a call to `Grizzly.restart_zipgateway`, that seems like something you'd _really_ want to know about.

Alternatively, we could add an on_exit callback to muontrap (or maybe just use the existing `exit_status_to_reason` option) to log an alert if Z/IP Gateway exits with a status other than 0.

Thoughts?